reverse-linked-list/Leo.py

```diff
@@ -0,0 +1,18 @@
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
+class Solution:
+    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
+        prev, curr = None, head
+
+        while curr:
+            tmp = curr.next ## save next node
+            curr.next = prev ## reverse next pointer to the prev
+            prev = curr ## update prev pointer with curr, since it's reversed now
+            curr = tmp ## move on to the next saved node
```

Comment:
```suggestion
        while curr:
            curr.next, prev, curr = prev, curr, curr.next
```

tmp 사용 없이 이렇게 하는 방법도 있을 것 같아요. ☺️

Reply:
감사합니다!! 포인터 문제들은 늘 할때마다 헷갈려서 tmp 해놓은거도 있고 안해놓은거도 있고 뒤죽박죽이네요 ㅎㅎ 좀 일관성있게 작성하도록 노력해보겠습니다 :D

Reply:
오 파이썬은 temp 없이 스왑이 되는군요...!